### PR TITLE
remove use of assigment for bmpi3 communicator

### PR DESCRIPTION
**Pull request by correaa**
## Proposed changes

remove artificial uses of deprecated BMPI3 functions, like assignments.

## What type(s) of changes does this code introduce?

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

Ubuntu 22.04

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- No. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by correaa** [comment]
did the test file because the coverage reduced (by 0.01%)? if so, i believe this is just because I removed lines (total number of lines reduced)

**Comment by markdewing** [comment]
Test this please

File: src/Message/Communicate.cpp [hunk 1]
```diff
@@ -37,10 +37,6 @@ Communicate* OHMMS::Controller = new Communicate;
 //default constructor: ready for a serial execution
 Communicate::Communicate() : myMPI(MPI_COMM_NULL), d_mycontext(0), d_ncontexts(1), d_groupid(0), d_ngroups(1) {}
 
-#ifdef HAVE_MPI
-Communicate::Communicate(const mpi3::environment& env) { initialize(env); }
```
Comment (markdewing) [inline src/Message/Communicate.cpp:41]: Should delete the corresponding declaration in Communicate.h

Comment (correaa) [reply src/Message/Communicate.cpp:41]: yes, please. That is correct.

**Review by markdewing — APPROVED** [review]

**Comment by markdewing** [comment]
Test this please

**Pull request merged**